tests/integration/targets/aci_interface_description/tasks/main.yml

```diff
@@ -0,0 +1,193 @@
+    <<: *aci_info
+    id: 1
+    state: query
+  register: version
+
+- name: Leaf - Clean test environment with no description interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: absent
+
+- name: Leaf - Query interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: query
+  register: query_leaf_description
+
+- name: Leaf - Verify that no infraHPathS is not present
+  assert:
+    that:
+    - query_leaf_description.current.0.infraInfra.children is not defined
+
+- name: Leaf - Set description on interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: present
+  register: set_leaf_description
+
+- name: Leaf - Verify that description is set on interface
+  assert:
+    that:
+    - set_leaf_description.current|length == 1
+    - set_leaf_description.current.0.infraHPathS.children.0.infraRsHPathAtt.attributes.tDn == "topology/pod-1/paths-1101/pathep-[eth1/33]"
+    - set_leaf_description.current.0.infraHPathS.attributes.descr == "testing"
+
+- name: Leaf - Clean test environment with no description interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: absent
+
+- name: Leaf - Query interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: query
+  register: query_leaf_description_again
+
+- name: Leaf - Verify that no infraHPathS is not present
+  assert:
+    that:
+    - query_leaf_description_again.current.0.infraInfra.children is not defined
+
+- name: Spine - Clean test environment with no description interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_spine
+    state: absent
+
+- name: Spine - Query interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_spine
+    state: query
+  register: query_spine_description
+
+- name: Spine - Verify that no infraHPathS is not present
+  assert:
+    that:
+    - query_spine_description.current.0.infraInfra.children is not defined
+
+- name: Spine - Set description on interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_spine
+    state: present
+  register: set_spine_description
+
+- name: Spine - Verify that description is set on interface
+  assert:
+    that:
+    - set_spine_description.current|length == 1
+    - set_spine_description.current.0.infraSHPathS.children.0.infraRsSHPathAtt.attributes.tDn == "topology/pod-1/paths-1201/pathep-[eth1/33]"
+    - set_spine_description.current.0.infraSHPathS.attributes.descr == "testing"
+
+- name: Spine - Clean test environment with no description interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_spine
+    state: absent
+
+- name: Spine - Query interface
+  cisco.aci.aci_interface_description:
+    <<: *aci_info
+    <<: *test_vars_leaf
+    state: query
```

Comment:
Is there any reason for using *test_vars_leaf instead of *test_vars_spine here?

Reply:
typo should be spine, will change